wdtk-storage/src/main/java/org/wikidata/wdtk/storage/datastructures/BitVectorImpl.java

```diff
@@ -162,16 +170,32 @@ public boolean addBit(boolean bit) {
 	 * @param position
 	 *            position
 	 * @throws IndexOutOfBoundsException
-	 *             if the position is out of bounds
+	 *             if the position is a negative number
 	 */
-	void assertRange(long position) throws IndexOutOfBoundsException {
-		if ((position < 0) || (position >= this.size)) {
+	void assertNonNegativePosition(long position)
+			throws IndexOutOfBoundsException {
+		if ((position < 0)) {
 			throw new IndexOutOfBoundsException("Position " + position
 					+ " is out of bounds.");
 		}
 	}
 
 	/**
+	 * Ensures that the bit vector is large enough to contain an element at the
+	 * given position. If the bit vector needs to be enlarged, new
+	 * <code>false</code> elements are added.
+	 * 
+	 * @param position
+	 *            position
+	 */
+	void ensureSize(long position) {
+		assertNonNegativePosition(position);
+		while (position >= this.size) {
+			addBit(false);
```

Comment:
Isn't this very inefficient? It seems that this loop might easily be executed 100 million times in a typical run. Can't we just increase the size in one step?